File: src/java/magma/app/compile/ConstructorHeader.java
package magma.app.compile;

import magma.app.MethodHeader;

public class ConstructorHeader implements MethodHeader<ConstructorHeader> {
    @Override
    public String generateWithAfterName(final String afterName) {
        return "constructor " + afterName;
    }

    @Override
    public boolean hasAnnotation(final String annotation) {
        return false;
    }

    @Override
    public ConstructorHeader removeModifier(final String modifier) {
        return this;
    }

    @Override
    public ConstructorHeader addModifier(final String modifier) {
        return this;
    }
}

File: src/node/magma/app/compile/ConstructorHeader.ts
import { ConstructorHeader } from "../../../magma/app/compile/ConstructorHeader";
import { MethodHeader } from "../../../magma/app/MethodHeader";
export class ConstructorHeader implements MethodHeader<ConstructorHeader> {
	generateWithAfterName(afterName: string): string {
		return "constructor " + afterName;
	}
	hasAnnotation(annotation: string): boolean {
		return false;
	}
	removeModifier(modifier: string): ConstructorHeader {
		return this;
	}
	addModifier(modifier: string): ConstructorHeader {
		return this;
	}
}
